src/main/cpp/src/hive_hash.cu

```diff
@@ -203,18 +203,64 @@ class hive_device_row_hasher {
     {
     }
 
-    template <typename T, CUDF_ENABLE_IF(not cudf::is_nested<T>())>
-    __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
-                                            cudf::size_type row_index) const noexcept
-    {
-      return this->hash_functor.template operator()<T>(col, row_index);
-    }
+    struct stack_element {
+      cudf::column_device_view col;
+      int child_index;
+      hive_hash_value_t hash;
+
+      __device__ stack_element() = delete;
+      __device__ stack_element(cudf::column_device_view col) : col(col), child_index(0), hash(HIVE_INIT_HASH) {}
+    };
 
-    template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
+    template <typename T>
     __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                             cudf::size_type row_index) const noexcept
     {
-      CUDF_UNREACHABLE("Nested type is not supported");
+      cudf::column_device_view curr_col = col.slice(row_index, 1);
+      stack_element stack[8] = {stack_element(curr_col), stack_element(curr_col), stack_element(curr_col), stack_element(curr_col),
```

Comment:
No need to initilize 8 column_device_view.
Consider:

```cpp
constexpr int len_of_8_stack_element = 8 * sizeof(stack_element);
// column_device_view default constructor is deleted, can not allocate column_device_view array directly
// use byte array to wrapper stack_element list
unit8 stack_wrapper[len_of_8_stack_element];
stack_element *stack = reinterpret_cast<stack_element*>(stack_wrapper);
```

Reply:
Please add some pseudocode to explain the calculation logic:
e.g.:
```cpp
// The stack is simulating the recursive function invokes:

hive_hash(float) {
   return (int)float;
}

hive_hash(list) {
  int hash = 0
  for(child : list) {
     // Depth first Traversal
     hash += hive_hash(child)
  }
  return hash
}
```

It's better to add an simple example.